meshapp/src/home/otp.tsx

```diff
@@ -0,0 +1,212 @@
+        timer: 2000,
+        timerProgressBar: true,
+        didOpen: () => {
+          Swal.showLoading();
+          const popup = Swal.getPopup();
+          if (popup !== null){
+            const timer = popup.querySelector("b");
+            if (timer !== null){
+              timerInterval = setInterval(() => {
+                let timeleft = Swal.getTimerLeft();
+                if(typeof timeleft === "number"){
+                  timeleft = Math.ceil(timeleft / 1000);
+                }
+                else{
+                  timeleft = 0;
+                }
+                timer.textContent = `${timeleft}`;
+              }, 100);
+            }
+          }
+        },
+        willClose: () => {
+          clearInterval(timerInterval);
+        }
+      }).then(() => {
+        navigate('/logged_out_home');
+      });
+    }
+    else {
+      const timeoutID = setTimeout(() => {  // Timer within OTP valid interval
+        let timerInterval: any;
+        Swal.fire({
+          title: "Timeout",
+          html: "Returning to login page in <b></b> seconds",
+          timer: 2000,
+          timerProgressBar: true,
+          didOpen: () => {
+            Swal.showLoading();
+            const popup = Swal.getPopup();
+            if (popup !== null){
+              const timer = popup.querySelector("b");
+              if (timer !== null){
+                timerInterval = setInterval(() => {
+                  let timeleft = Swal.getTimerLeft();
+                  if(typeof timeleft === "number"){
+                    timeleft = Math.ceil(timeleft / 1000);
+                  }
+                  else{
+                    timeleft = 0;
+                  }
+                  timer.textContent = `${timeleft}`;
+                }, 100);
+              }
+            }
+          },
+          willClose: () => {
+            clearInterval(timerInterval);
+          }
+        }).then(() => {
+          navigate('/logged_out_home'); // Return to login page if OTP isn't validated within interval
+          removeCookie("user_id");
+        });
+      }, 60000);  // OTP interval in ms
+      return () => {
+        clearTimeout(timeoutID);  // stop timer if OTP is valid
+      };
+    }
+  }, [])
+
+  const handleSubmit = async(event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    data.append('otp', otp)
+    const response = await axiosInstance.post('accounts/verify-two-factor-auth/',{
+      "accountID":cookies.user_id,
+      "otp":data.get("otp")
+    })
+      .then((axiosResponse) => {
+        Swal.fire({
+          position: "center",
+          icon: "success",
+          title: "Verified!",
+          showConfirmButton: false,
+          timer: 1500
+        });
+        removeCookie("user_id");
+        navigate('/logged_in_home');
+      })
+      .catch(errorHandler); // User's OTP could not be verified
+  }
```

Comment:
I would figure that something similar as to what I mentioned earlier about try/catch blocks instead of callbacks with .then and .catch could be used here.

Reply:
👍🏼